daemon/execdriver/windows/run.go

```diff
@@ -0,0 +1,271 @@
+// +build windows
+
+package windows
+
+// Note this is alpha code for the bring up of containers on Windows.
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/daemon/execdriver"
+	"github.com/microsoft/hcsshim"
+	"github.com/natefinch/npipe"
+)
+
+func (d *driver) Run(c *execdriver.Command, pipes *execdriver.Pipes, startCallback execdriver.StartCallback) (execdriver.ExitStatus, error) {
+
+	var (
+		term                           execdriver.Terminal
+		err                            error
+		inListen, outListen, errListen *npipe.PipeListener
+	)
+
+	// Make sure the client isn't asking for options which aren't supported
+	err = checkSupportedOptions(c)
+	if err != nil {
+		return execdriver.ExitStatus{ExitCode: -1}, err
+	}
+
+	type layer struct {
+		Id   string
+		Path string
+	}
+
+	type defConfig struct {
+		DefFile string
+	}
+
+	type networkConnection struct {
+		NetworkName string
+		EnableNat   bool
+	}
+	type networkSettings struct {
+		MacAddress string
+	}
+
+	type device struct {
+		DeviceType string
+		Connection interface{}
+		Settings   interface{}
+	}
+
+	type containerInit struct {
+		SystemType              string
+		Name                    string
+		IsDummy                 bool
+		VolumePath              string
+		Devices                 []device
+		IgnoreFlushesDuringBoot bool
+		LayerFolderPath         string
+		Layers                  []layer
+	}
```

Comment:
any particular reason these types are embedded in the function rather than at the package level?  I realize at this moment they are private, so I assume it doesn't necessarily matter, but makes this function even longer which is already pretty long..